apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -46,19 +48,45 @@ public abstract partial class BigtableClient
         /// "default" application profile will be used.
         /// </param>
         /// <returns>The task representing the created <see cref="BigtableClient"/>.</returns>
-        public static async Task<BigtableClient> CreateAsync(ServiceEndpoint endpoint = null, BigtableServiceApiSettings settings = null, string appProfileId = null)
+        public static async Task<BigtableClient> CreateAsync(BigtableServiceApiSettings settings = null, ClientCreationSettings clientCreationSettings = null, string appProfileId = null)
         {
-            var client = await BigtableServiceApiClient.CreateAsync(endpoint, settings).ConfigureAwait(false);
-            return new BigtableClientImpl(client, appProfileId) { UnderlyingClientSettings = settings };
+            clientCreationSettings?.Validate();
+            var channelCredentials = clientCreationSettings?.Credentials;
+            // Use default credentials if none given.
+            if (channelCredentials == null)
+            {
+                var credentials = await GoogleCredential.GetApplicationDefaultAsync().ConfigureAwait(false);
+                if (credentials.IsCreateScopedRequired)
+                {
+                    credentials = credentials.CreateScoped(BigtableServiceApiClient.DefaultScopes);
+                }
+                channelCredentials = credentials.ToChannelCredentials();
+            }
+            var clientCount = clientCreationSettings?.ClientCount ?? Environment.ProcessorCount;
+
+            var endpoint = clientCreationSettings?.ServiceEndpoint ?? BigtableServiceApiClient.DefaultEndpoint;
+            var clients = new BigtableServiceApiClient[clientCount];
+            var shutdowns = new Func<Task>[clientCount];
+            // Fill clients[] with BigtableServiceApiClient instances, each with specific channel
+            for (int i = 0; i < clientCount; i++)
+            {
+                var channel = new Channel(endpoint.Host, endpoint.Port, channelCredentials);
```

Comment:
This doesn't pass the `channelOptions` like Pubsub does. Was that error with message sizes Pubsub specific?

Reply:
I have only saw [this](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.cs#L260) documentation in PubSub. I did not pass the `channelOptions` when implemented `BigtableClientPool` in a test project when tested channel pooling effect on performance. I followed [these](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/docs/root/faq.md#how-can-i-use-non-default-credentials-for-grpc-based-apis) instructions to create a `BigtableClient` with a specified channel.

Reply:
@chrisdunelm why was this needed in Pubsub? Is this something we need to be concerned about for Bigtable as well?

Reply:
Is this referring to the code in [here](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.cs#L242)?

If so, then this is because it is expected that both publish and subscriber messages will exceed the default maximum individual message size allowed client-side (~4Mb). The servre-side maximum for pubsub is ~10Mb. The client is set to -1 indicating that both sent and received messages can be of any size.

Whether these particular channel options are needed here depends on the maximum individual message size you expect to send and receive on the client.

Reply:
Ok thanks Chris, I'll ask internally.